### PR TITLE
[FR] v2.1 implementation 

### PR DESCRIPTION
**Pull request by mssfang**
- Implementation folder regenerated abased swagger v2.1. Ignore all changes in the ***Implementation*** folder. 
- Removed gender related classes
- Add more static values to `FormRecognizerLanguage` expandable string class.
- Removed the `pollInterval` property from all endpoints' options bag, such as `RecognizeBusinessCardsOptions`, etc. Polling interval
  can be updated in the Azure Core `SyncPoller` or `PollerFlux`'s method, `setPollInterval()`, synchronously and asynchronously, respectively.
- Make `FormLine`, `FormPage`, `FormTable`, `FormSelectionMark`, `TextAppearance`, `CustomFormModel`, `CustomFormModelInfo`, `CustomFormModelProperties`
  `CustomFormSubmodel`, `TrainingDocumentInfo` model class immutable.
  
- fixes the first issue in https://github.com/Azure/azure-sdk-for-java/issues/18967  

  API view: https://apiview.dev/Assemblies/Review/1e7f0076510141049dc41cd0e65d8c8d?diffRevisionId=7c27da079c024e0aafecacddc20dca04&doc=False&diffOnly=False

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -55,7 +55,7 @@
 /sdk/eventgrid/                                     @srnagar @yijunxiems
 
 # PRLabel: %Event Hubs
-/sdk/eventhubs/                                     @conniey @srnagar @mssfang @yijunxiems
+/sdk/eventhubs/                                     @conniey @srnagar @yijunxiems
```
Comment (mssfang) [inline .github/CODEOWNERS:58]: this changes will be disappear after I merge https://github.com/Azure/azure-sdk-for-java/pull/21245

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
-## 3.1.0-beta.4 (Unreleased)
+## 3.1.0 (Unreleased)
+### Features Added
+- Added `clientOptions()` and `getDefaultLogOptions()` methods to the `FormRecognizerClientBuilder` and `FormTrainingClientBuilder`.
+- We are able to support multiple service API versions now: `V2_0` and `V2_1`.
+- Add more static values to `FormRecognizerLanguage` expandable string class.
+
 ### Breaking Changes
-- Renamed `ReadingOrder` model to `FormReadingOrder`.
+- The client defaults to the latest supported service version, which currently is 2.1.
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:10]: should the removal of FieldValueType "gender" and "country" and addition of "countryRegion" be added here?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java [hunk 3]
```diff
@@ -402,13 +402,9 @@ private static FormField setFormField(String name, FieldData valueData, FieldVal
...
                     FieldValueType.SELECTION_MARK_STATE);
                 break;
-            case GENDER:
-                value = new com.azure.ai.formrecognizer.models.FieldValue(fieldValue.getValueGender(),
-                    FieldValueType.GENDER);
-                break;
-            case COUNTRY:
-                value = new com.azure.ai.formrecognizer.models.FieldValue(fieldValue.getValueCountry(),
-                    FieldValueType.COUNTRY);
+            case COUNTRY_REGION:
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/Transforms.java:405]: Not specifically this line, but above where we have the workaround to use the `text` value for the selection mark - you should be able to use the valueSelectionMark now that the service fixed the bug in v2.1

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java [hunk 4]
```diff
@@ -1077,7 +1077,7 @@ public void beginRecognizeIdentityDocumentsFromUrlWithOptions() {
                     FormField countryFormField = recognizedFields.get("Country");
                     if (countryFormField != null) {
                         if (FieldValueType.STRING == countryFormField.getValue().getValueType()) {
-                            String country = countryFormField.getValue().asCountry();
+                            String country = countryFormField.getValue().asCountryRegion();
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java:1080]: might be worth updating the variable name to countryRegionFormField as well

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeIdentityDocumentsFromUrl.java [hunk 5]
```diff
@@ -30,12 +30,12 @@
     public static void main(final String[] args) throws IOException {
         // Instantiate a client that will be used to call the service.
         FormRecognizerClient client = new FormRecognizerClientBuilder()
-            .credential(new AzureKeyCredential("{key}"))
-            .endpoint("https://{endpoint}.cognitiveservices.azure.com/")
+            .credential(new AzureKeyCredential("3e571e46580141aaae155f0099a5a54b"))
```
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeIdentityDocumentsFromUrl.java:33]: is this a real key?

Comment (kristapratico) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeIdentityDocumentsFromUrl.java:33]: I don't see you online right now so I rotated the key for `mariari-canada-central`. Please check if anything else needs to be done to properly scrub the secret.

Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/RecognizeIdentityDocumentsFromUrl.java]: Sorry. Will follow https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/101/Leaked-secret-procedure

**Review by alzimmermsft — APPROVED** [review]
Looks good from the perspective of versions

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java [hunk 6]
```diff
@@ -1077,7 +1077,7 @@ public void beginRecognizeIdentityDocumentsFromUrlWithOptions() {
                     FormField countryFormField = recognizedFields.get("Country");
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java:1077]: this should be `CountryRegion`

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java [hunk 7]
```diff
@@ -947,7 +947,7 @@ public void beginRecognizeIdentityDocumentsFromUrl() {
                 FormField countryFormField = recognizedFields.get("Country");
```
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java:947]: same here. Anywhere where u had `Country` triple check that it is now `CountryRegion`

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerClientJavaDocCodeSnippets.java:947]: applies to all samples and tests

**Review by kristapratico — APPROVED** [review]
changes LGTM

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java [hunk 8]
```diff
@@ -686,7 +687,7 @@
             final boolean isFieldElementsIncluded = finalRecognizeBusinessCardsOptions.isFieldElementsIncluded();
             final FormRecognizerLocale localeInfo = finalRecognizeBusinessCardsOptions.getLocale();
             return new PollerFlux<>(
-                finalRecognizeBusinessCardsOptions.getPollInterval(),
+                DEFAULT_POLL_INTERVAL,
```
Comment (witemple-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:690]: Am I understanding this correctly that this makes the polling interval not configurable?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java [hunk 9]
```diff
@@ -228,32 +223,17 @@ public SelectionMarkState asSelectionMarkState() {
...
-        }
-        return this.formFieldCountry;
-    }
-
-    /**
-     * Gets the value of the field as a gender value.
+     * Gets the value of the field as a COUNTRY_REGION field.
      *
-     * @return the value of the field as a gender value.
+     * @return the value of the field as COUNTRY_REGION.
```
Comment (witemple-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/models/FieldValue.java:228]: using COUNTRY_REGION seems strange here, maybe "as a country or region in the world"

File: sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java [hunk 10]
```diff
@@ -1005,12 +1005,12 @@ public void beginRecognizeIDDocumentFromUrl() {
...
-                    if (countryFormField != null) {
-                        if (FieldValueType.STRING == countryFormField.getValue().getValueType()) {
-                            String country = countryFormField.getValue().asCountry();
-                            System.out.printf("Country: %s, confidence: %.2f%n",
-                                country, countryFormField.getConfidence());
+                    FormField countryRegionFormField = recognizedFields.get("CountryRegion");
+                    if (countryRegionFormField != null) {
+                        if (FieldValueType.STRING == countryRegionFormField.getValue().getValueType()) {
+                            String countryRegion = countryRegionFormField.getValue().asCountryRegion();
+                            System.out.printf("Country region: %s, confidence: %.2f%n",
```
Comment (witemple-msft) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/samples/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClientJavaDocCodeSnippets.java:1012]: I would stick to the same rendering as the service uses here, or expand to "Country or region".

```suggestion
                            System.out.printf("CountryRegion: %s, confidence: %.2f%n",
```

**Review by witemple-msft — COMMENTED** [review]
Couple of observations.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java (hunk 8 above)
Comment (mssfang) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java:690]: it set the default value when creating poller, but we can override the value by using Poller's method `setPollInterval()`

**Pull request merged**